azure/services/aso/service.go

```diff
@@ -31,6 +32,9 @@ type Service[T deepCopier[T], S Scope] struct {
 
 	Scope S
 	Specs []azure.ASOResourceSpecGetter[T]
+	// ListFunc is used to enumerate ASO resources to determine which stale ones to delete. This behavior may
```

Comment:
Does this list interface only have one usage, to cleanup existent resources that no longer have an active CAPZ reference? If so is there a more specific name that would be better?

Or would it be better to make the comment more like:

```golang
// ListFunc is used to enumerate ASO exostent resources.
// Currently this interface is designed only to aid discovery of ASO resources
// that no longer have a CAPZ reference, and can thus be deleted. This behavior may
// be skipped for a service by leaving this field nil.
```

I can see why we want to generalize this as `ListFunc`, but it's a bit non-intuitive that we're only employing it for this very specific reason (to list and then garbage collect).

Reply:
> Does this list interface only have one usage, to cleanup existent resources that no longer have an active CAPZ reference?

Yes


> Or would it be better to make the comment more like:
> 
> ```go
> // ListFunc is used to enumerate ASO exostent resources.
> // Currently this interface is designed only to aid discovery of ASO resources
> // that no longer have a CAPZ reference, and can thus be deleted. This behavior may
> // be skipped for a service by leaving this field nil.
> ```

I think this helps, so I'll update the comment here. I think it's worth trying to avoid having separate `ListForCleanup` and `ListForXYZ` if they do the same thing.

Reply:
+1 to the clarifying comment, and yes making the name too specific isn't the right approach for an interface method (that may be used for a variety of use cases in the future)